src/main/java/com/autotune/experimentManager/data/input/metrics/EMMetricResult.java

```diff
@@ -24,7 +24,9 @@ public EMMetricResult(JSONObject jsonObject) throws IncompatibleInputJSONExcepti
             isPercentileResultsAvailable = true;
         }
         emMetricGenericResults = new EMMetricGenericResults(jsonObject.getJSONObject(EMConstants.EMJSONKeys.GENERAL_INFO));
-        emMetricPercentileResults = new EMMetricPercentileResults(jsonObject.getJSONObject(EMConstants.EMJSONKeys.PERCENTILE_INFO));
+        if (isPercentileResultsAvailable){
+            emMetricPercentileResults = new EMMetricPercentileResults(jsonObject.getJSONObject(EMConstants.EMJSONKeys.PERCENTILE_INFO));
+        }
```

Comment:
Please do the following change instead of above
```
-        if (!jsonObject.has(AutotuneConstants.JSONKeys.GENERAL_INFO) ||
+        if (!jsonObject.has(AutotuneConstants.JSONKeys.GENERAL_INFO) &&
             !jsonObject.has(AutotuneConstants.JSONKeys.PERCENTILE_INFO)) {
             throw new IncompatibleInputJSONException();
         }
+        emMetricGenericResults = new EMMetricGenericResults(jsonObject.getJSONObject(AutotuneConstants.JSONKeys.GENERAL_INFO));
         if (jsonObject.has(AutotuneConstants.JSONKeys.PERCENTILE_INFO)) {
             isPercentileResultsAvailable = true;
+            emMetricPercentileResults = new EMMetricPercentileResults(jsonObject.getJSONObject(AutotuneConstants.JSONKeys.PERCENTILE_INFO));
         }
-        emMetricGenericResults = new EMMetricGenericResults(jsonObject.getJSONObject(AutotuneConstants.JSONKeys.GENERAL_INFO));
-        emMetricPercentileResults = new EMMetricPercentileResults(jsonObject.getJSONObject(AutotuneConstants.JSONKeys.PERCENTILE_INFO));
```
The above works for me

Reply:
Please also add getters for both the objects

Reply:
Ah! sorry for dumb change made. Not sure how i missed it, might be an overlook. Fixed it now